cogeo_mosaic/backends/base.py

```diff
@@ -18,18 +19,57 @@
 
 from cogeo_mosaic.backends.utils import find_quadkeys, get_hash
 from cogeo_mosaic.cache import cache_config
-from cogeo_mosaic.errors import NoAssetFoundError
+from cogeo_mosaic.errors import NoAssetFoundError, UnsupportedOperation
 from cogeo_mosaic.models import Info, Metadata
 from cogeo_mosaic.mosaic import MosaicJSON
 from cogeo_mosaic.utils import bbox_union
 
 
+def update_zoom_and_bounds(instance, attribute, value):
+    """Update Zooms and Bounds attribute when mosaic_def is set."""
+    instance.minzoom = value.minzoom
+    instance.maxzoom = value.maxzoom
+    instance.bounds = value.bounds
+    return value
+
+
+def mode_validator(instance, attribute, value):
+    """Validate mode."""
+    if value not in instance._available_modes:
+        raise ValueError(
+            f"Invalid '{value}' mode. MUST be one of {instance._available_modes}."
+        )
+
+    if "r" in value:
+        instance._readable = True
+
+    elif "w" in value:
+        instance._writable = True
+
+    if "+" in value:
+        instance._writable = True
```

Comment:
Inspiration from https://github.com/python/cpython/blob/3.9/Lib/_pyio.py

Reply:
You should also add 
```
        instance._readable = True
```

This works:
```py
with open('/tmp/test.py', 'w+') as f:
    f.read()
```